week3/project/ecommerce-shop-project/src/components/Site.jsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from "react";
+import Categories from "./Categories";
+import Products from "./Products";
+import useCategories from "./useCategories.js"
+import useProducts from "./useProducts.js"
+// import { Link } from "react-router-dom";
+// import ProductDetail from "./ProductDetail";
+// import Product from "./Product";
+
+
+function SiteWorking() {
+  const [selectedCategory, setSelectedCategory] = useState("");
+  
+  const {categories, loading: categoriesLoading, error:categoriesError} =
+      useCategories();
+    const {products, loading: productsLoading, error: productsError} =
+      useProducts();
```

Comment:
perfect custom hooks but from inside you are using only useFetch and returning the data which comes from the url so its the same as you are using useFetch directly

Reply:
What would you propose/advise to add there? Maybe I don't get the idea of custom hook completely(((